manage

```diff
@@ -353,6 +358,226 @@ $app"
+        secondary_write_list_pid=$!
+      fi
+      
+      #run manage script for app installs, uninstalls, or updates. Avoid using it for file-updates and refreshes because that is out of the scope for manage script.
+      if [ "$action" == update-file ];then
+        #Set terminal title
+        echo -ne "\e]0;Updating ${app} file\a"
+        update_file "$app"
+        exitcode=$?
+      elif [ "$action" == refresh ];then
+        #Set terminal title
+        echo -ne "\e]0;Refreshing ${app}\a"
+        refresh_app "$app"
+        exitcode=$?
+      elif [ "$action" == update ];then #manage can handle this action, but avoid spawning subprocess
+        #Set terminal title
+        echo -ne "\e]0;Updating ${app}\a"
+        update_app "$app"
+        exitcode=$?
+      else
+        #Set terminal title
+        echo -ne "\e]0;${action^}ing ${app}\a"
+        "${DIRECTORY}/manage" "$action" "$app"
+        exitcode=$?
+      fi
+      
+      #record exit code in current line of $queue
+      [ "$exitcode" != 0 ] && exitcode=1 #for easier regex and sed lines, force non-zero exit codes to be 1
+      queue="$(echo "$queue" | sed "${current_line_num}s/;in-progress$/;$exitcode/")"
+      
+      #one more line of $queue has been completed.
+      current_line_num=$((current_line_num+1))
+    done
+    
+    #Daemon has stopped listening, so remove its pid file
+    rm -f "${DIRECTORY}/data/manage-daemon/pid"
+    
+    #refresh the list in queue-viewer window for the final time with all actions complete
+    [ ! -z "$secondary_write_list_pid" ] && kill "$secondary_write_list_pid" 2>/dev/null #kill secondary list writer
+    wait $write_list_pid
+    write_list "$queue"
+    
+    #before exiting the loop, add a line to the queue-viewer window indicating that all items have completed.
+    echo "${DIRECTORY}/icons/none-1.png
+${DIRECTORY}/icons/none-1.png
+Done.
+${DIRECTORY}/icons/none-1.png
+" > "${DIRECTORY}/data/manage-daemon/yadlist"
+    
+    #close the queue-viewer window in a few seconds
+    (sleep 5; kill $yadpid 2>/dev/null) &
+    
+    #display "all done" window where the queue was
+    export -f summary_window
+    setsid bash -c summary_window
+  }
+  
+  summary_window() { #given $queue, summarize what was done
+    local IFS=$'\n'
+    export botspot_message='to Botspot (Pi-Apps founder, college student, nearly bankrupt)'
+    export gman_message='to theofficialgman (notable Pi-Apps contributor)'
+    summary_window_click_function() {
+      if [ "$5" == "$botspot_message" ];then
+        xdg-open 'https://github.com/sponsors/botspot' &
+        
+        #remember that donation link was clicked - opts out of any popups introduced in the future
+        #"$(echo 'botspot donation click' | sha1sum | awk '{print $1}')"
+        if ! grep -q "8a780b08ac20332b897ecb399b25f4622bb31341" "${DIRECTORY}/data/runonce_hashes" ;then
+          echo "8a780b08ac20332b897ecb399b25f4622bb31341" >> "${DIRECTORY}/data/runonce_hashes"
+        fi
```

Comment:
is this dead code?

Reply:
no, the commented code just documents what the otherwise random looking hash represents.

Reply:
That was actually one of my most recent changes. The idea is if in the future I add some kind of donation message elsewhere with a popup, it won't appear for everybody who already clicked donate in the past.